### PR TITLE
K8SPSMDB-355 update doc how to scale cluster

### PR DESCRIPTION
**Pull request by fiowro**
[K8SPSMDB-355](https://jira.percona.com/browse/K8SPSMDB-355) [&#10088;?&#10089;](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

### CODE REVIEW

File: source/scaling.rst [hunk 1]
```diff
@@ -6,21 +6,45 @@ platform is the ease of an application scaling. Scaling a Deployment up
 or down ensures new Pods are created and set to available Kubernetes
 nodes.
 
-Size of the cluster is controlled by a :ref:`size key<replsets-size>` in the :ref:`operator.custom-resource-options` configuration.. That’s why scaling the cluster needs
-nothing more but changing this option and applying the updated
-configuration file. This may be done in a specifically saved config, or
-on the fly, using the following command, which saves the current
-configuration, updates it and applies the changed version:
+The size of the cluster is controlled by ``size`` keys in the
```
Comment (spron-in) [inline source/scaling.rst:9]: Should it be ``size`` key, not key**s**?

File: source/scaling.rst [hunk 2]
```diff
@@ -6,21 +6,45 @@ platform is the ease of an application scaling. Scaling a Deployment up
...
-Percona Server for MongoDB is not recommended, as it makes ``size``
-configuration option out of sync, and the next config change may result
-in reverting the previous number of nodes.*
+Scaling the cluster down is done in a similar way as scaling it up. The only
+thing to consider when you scale your cluster down is how the Operator treats
+`Persistent Volume Claims <https://kubernetes.io/docs/concepts/storage/persistent-volumes/>`_.
+By default, PVCs are kept after the database deletion. But you can enable
+automatic removal of PVCs after the database scale down or deletion event by
+adding the ``metadata.finalizers.delete-psmdb-pvc`` key in the
+``deploy/cr.yaml`` configuration file.
```
Comment (spron-in) [inline source/scaling.rst:50]: @blez is scaling handled by finalizer in this release?

File: source/scaling.rst [hunk 3]
```diff
@@ -6,21 +6,45 @@ platform is the ease of an application scaling. Scaling a Deployment up
...
+
+.. code:: yaml
+
+   ....
+   replsets:
+     ....
+     size: 5
+     ....
+
+Don't forget to present changed configuration to the Operator as usual, running
```
Comment (spron-in) [inline source/scaling.rst:39]: Could you please review this line? 

"Don't forget to present changed configuration to the Operator as usual" sounds weird.

File: source/operator.rst [hunk 4]
```diff
@@ -4,8 +4,15 @@ Custom Resource options
...
 The operator is configured via the spec section of the
-`deploy/cr.yaml <https://github.com/percona/percona-server-mongodb-operator/blob/main/deploy/cr.yaml>`_
-file. This file contains the following spec sections:
+`deploy/cr.yaml <https://github.com/percona/percona-server-mongodb-operator/blob/main/deploy/cr.yaml>`_ file.
+
+The metadata part of this file contains the following keys:
+
+* ``name`` (``my-cluster-name`` by default) sets the name of your Percona Server
+  for MongoDB Cluster,
+* ``finalizers.delete-psmdb-pvc``, if present, activates the `Finalizer <https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#finalizers>`_ which deletes appropriate `Persistent Volume Claims <https://kubernetes.io/docs/concepts/storage/persistent-volumes/>`_ after the database scale down or deletion event (off by default).
```
Comment (spron-in) [inline source/operator.rst:13]: I'm not sure about the scaling.

File: source/scaling.rst (hunk 2 above)
Comment (blez) [reply source/scaling.rst:50]: We have finalizer for cluster deletion, not scaling down
@spron-in

**Review by spron-in — APPROVED** [review]

File: source/scaling.rst [hunk 5]
```diff
@@ -6,21 +6,35 @@ platform is the ease of an application scaling. Scaling a Deployment up
...
-on the fly, using the following command, which saves the current
-configuration, updates it and applies the changed version:
+The size of the cluster is controlled by the ``size`` key in the
+:ref:`operator.custom-resource-options` configuration.
 
-.. code:: bash
+.. note:: Using ``kubectl scale StatefulSet_name`` command to rescale
+   Percona Server for MongoDB is not recommended, as it makes ``size``
+   configuration options out of sync, and the next config change may result
+   in reverting the previous number of nodes.
```
Comment (spron-in) [inline source/scaling.rst:15]: This is not correct.
1. Operator will not allow the user to scale statefulset and will revert to the original size in the next reconcile loop (5 seconds).
2. The command for scaling StatefulSet is: kubectl scale statefulset StatefulSet_name

**Review by spron-in — CHANGES_REQUESTED** [review]

File: source/scaling.rst [hunk 6]
```diff
@@ -6,21 +6,34 @@ platform is the ease of an application scaling. Scaling a Deployment up
...
-configuration file. This may be done in a specifically saved config, or
-on the fly, using the following command, which saves the current
-configuration, updates it and applies the changed version:
+The size of the cluster is controlled by the ``size`` key in the
+:ref:`operator.custom-resource-options` configuration.
 
-.. code:: bash
+.. note:: The Operator will not allow rescaling Percona Server for MongoDB with
+   the ``kubectl scale statefulset <StatefulSet name>`` command because this
+   would cause ``size`` configuration options to become out of sync.
```
Comment (spron-in) [inline source/scaling.rst:14]:
```suggestion
.. note:: The Operator will not allow to scale Percona Server for MongoDB with
   the ``kubectl scale statefulset <StatefulSet name>`` command as it puts
``size`` configuration options out of sync.
```

**Review by spron-in — DISMISSED** [review]

**Review by tplavcic — APPROVED** [review]

**Review by spron-in — APPROVED** [review]

**Pull request merged**